File: src/lib/seeds.ts
/**
 * Hand-picked starting points for a rabbit hole.
 *
 * Wikipedia's `list=random` returns mostly obscure stubs ("Uniform Securities Act"),
 * which makes for a terrible first card. These are dense, image-rich, heavily-linked
 * articles that fan out into interesting territory — a much better cold start.
 *
 * Each seed files under a mood category so the start page can offer "I'm in the
 * mood for animals" as an entry point, not just an undifferentiated chip pile.
 */
export type SeedCategory = 'history' | 'animals' | 'geography' | 'culture' | 'space' | 'science';

export interface Seed {
	title: string;
	category: SeedCategory;
}

/** Display order for the category row (Ben's five, plus Science — the seed list
 *  is a third technology/science and filing those under Culture reads wrong). */
export const SEED_CATEGORIES: readonly { id: SeedCategory; label: string }[] = [
	{ id: 'history', label: 'History' },
	{ id: 'animals', label: 'Animals' },
	{ id: 'geography', label: 'Geography' },
	{ id: 'culture', label: 'Culture' },
	{ id: 'space', label: 'Space' },
	{ id: 'science', label: 'Science' }
];

export const SEEDS: Seed[] = [
	{ title: 'Roman Empire', category: 'history' },
	{ title: 'Silk Road', category: 'history' },
	{ title: 'Chernobyl disaster', category: 'history' },
	{ title: 'Library of Alexandria', category: 'history' },
	{ title: 'Samurai', category: 'history' },
	{ title: 'Pyramids of Giza', category: 'history' },
	{ title: 'Vikings', category: 'history' },
	{ title: 'Spice trade', category: 'history' },
	{ title: 'Aztecs', category: 'history' },
	{ title: 'Antikythera mechanism', category: 'history' },
	{ title: 'Octopus', category: 'animals' },
	{ title: 'Tardigrade', category: 'animals' },
	{ title: 'Great Barrier Reef', category: 'animals' },
	{ title: 'Bioluminescence', category: 'animals' },
	{ title: 'Honey bee', category: 'animals' },
	{ title: 'Volcano', category: 'geography' },
	{ title: 'Antarctica', category: 'geography' },
	{ title: 'Deep sea', category: 'geography' },
	{ title: 'Plate tectonics', category: 'geography' },
	{ title: 'Mount Everest', category: 'geography' },
	{ title: 'Mariana Trench', category: 'geography' },
	{ title: 'Jazz', category: 'culture' },
	{ title: 'Renaissance', category: 'culture' },
	{ title: 'Coffee', category: 'culture' },
	{ title: 'Chess', category: 'culture' },
	{ title: 'Hokusai', category: 'culture' },
	{ title: 'Black hole', category: 'space' },
	{ title: 'Fermi paradox', category: 'space' },
	{ title: 'Aurora', category: 'space' },
	{ title: 'Voyager 1', category: 'space' },
	{ title: 'Saturn', category: 'space' },
	{ title: 'Mycology', category: 'science' },
	{ title: 'Cryptography', category: 'science' },
	{ title: 'Quantum mechanics', category: 'science' },
	{ title: 'Human brain', category: 'science' },
	{ title: 'Internet', category: 'science' },
	{ title: 'Penicillin', category: 'science' }
];

/** Pick a random curated seed (for "Surprise me"), optionally within one category. */
export function randomSeed(category?: SeedCategory): Seed {
	const pool = category ? SEEDS.filter((s) => s.category === category) : SEEDS;
	return pool[Math.floor(Math.random() * pool.length)];
}

File: ios/Tangent/Models/Seeds.swift
import Foundation

/// Hand-picked starting points — dense, heavily-linked articles that fan out into
/// interesting territory (Wikipedia's random endpoint mostly returns dull stubs).
/// A subset of the web `SEEDS`.
struct Seed: Hashable {
	let title: String
	let emoji: String
}

enum Seeds {
	static let all: [Seed] = [
		Seed(title: "Roman Empire", emoji: "🏛️"),
		Seed(title: "Black hole", emoji: "🕳️"),
		Seed(title: "Octopus", emoji: "🐙"),
		Seed(title: "Silk Road", emoji: "🐫"),
		Seed(title: "Chernobyl disaster", emoji: "☢️"),
		Seed(title: "Jazz", emoji: "🎷"),
		Seed(title: "Volcano", emoji: "🌋"),
		Seed(title: "Library of Alexandria", emoji: "📜"),
		Seed(title: "Mycology", emoji: "🍄"),
		Seed(title: "Antarctica", emoji: "🧊"),
		Seed(title: "Cryptography", emoji: "🔐"),
		Seed(title: "Renaissance", emoji: "🎨"),
		Seed(title: "Deep sea", emoji: "🌊"),
		Seed(title: "Samurai", emoji: "⚔️"),
		Seed(title: "Quantum mechanics", emoji: "⚛️"),
		Seed(title: "Coffee", emoji: "☕"),
		Seed(title: "Fermi paradox", emoji: "👽"),
		Seed(title: "Vikings", emoji: "🛶"),
		Seed(title: "Human brain", emoji: "🧠"),
		Seed(title: "Chess", emoji: "♟️"),
		Seed(title: "Tardigrade", emoji: "🐻"),
		Seed(title: "Aztecs", emoji: "🗿")
	]

	/// A deterministic-enough starting seed for cold launch (varies by day).
	static func cold() -> Seed {
		let day = Calendar.current.ordinality(of: .day, in: .era, for: Date()) ?? 0
		return all[day % all.count]
	}
}
